packages/notifications/lib/apn.js

```diff
@@ -58,12 +60,22 @@ const publish = async (args) => {
         ? { expiry: parseInt(new Date(now.getTime() + now.getTimezoneOffset() + ttl).getTime() / 1000) }
         : { }
     })
-    debug('sending %d notifications...', tokens.length)
-    return Promise.all(
+    const result = await Promise.all(
       tokens.map(token =>
         provider.send(message, token)
       )
     )
+    debug('APN: #recipients %d, message: %O, result: %O', tokens.length, message, result)
+    const staleTokens = result.failed.reduce((acc, cur) => {
```

Comment:
where does `result.failed` come from? AFAIK `Promise.all` does not return such a thing.

Reply:
That's true, I have to reduce the `result` array first…